spec/features/sales_log_spec.rb

```diff
@@ -190,141 +190,153 @@
-
     before do
-      allow(user).to receive(:need_two_factor_authentication?).and_return(false)
-      sign_in user
-      sales_log.update!(purchid: "different")
-      visit("/sales-logs/#{sales_log.id}/purchaser-code")
-      fill_in("sales-log-purchid-field", with: duplicate_log.purchid)
-      click_button("Save and continue")
+      Timecop.freeze(Time.zone.local(2024, 3, 3))
+      Singleton.__init__(FormHandler)
     end
 
-    it "allows keeping the original log and deleting duplicates" do
-      sales_log.reload
-      duplicate_log.reload
-      expect(sales_log.duplicates.count).to eq(1)
-      expect(duplicate_log.duplicates.count).to eq(1)
-      expect(sales_log.duplicate_set_id).not_to be_nil
-      expect(duplicate_log.duplicate_set_id).not_to be_nil
-      expect(sales_log.duplicates).to include(duplicate_log)
-
-      expect(page).to have_current_path("/sales-logs/#{sales_log.id}/duplicate-logs?original_log_id=#{sales_log.id}")
-      click_link("Keep this log and delete duplicates", href: "/sales-logs/#{sales_log.id}/delete-duplicates?original_log_id=#{sales_log.id}")
-      expect(page).to have_current_path("/sales-logs/#{sales_log.id}/delete-duplicates?original_log_id=#{sales_log.id}")
-      click_button "Delete this log"
-      duplicate_log.reload
-      expect(duplicate_log.deleted?).to be true
-      expect(page).to have_css(".govuk-notification-banner.govuk-notification-banner--success")
-      expect(page).to have_content("Log #{duplicate_log.id} has been deleted.")
-      expect(page).to have_current_path("/sales-logs/#{sales_log.id}/duplicate-logs?organisation_id=&original_log_id=#{sales_log.id}&referrer=")
-      expect(page).not_to have_content("These logs are duplicates")
-      expect(page).not_to have_link("Keep this log and delete duplicates")
-      expect(page).to have_link("Back to Log #{sales_log.id}", href: "/sales-logs/#{sales_log.id}")
-
-      sales_log.reload
-      duplicate_log.reload
-
-      expect(sales_log.duplicates.count).to eq(0)
-      expect(duplicate_log.duplicates.count).to eq(0)
-      expect(sales_log.duplicate_set_id).to be_nil
-      expect(duplicate_log.duplicate_set_id).to be_nil
+    after do
+      Timecop.return
+      Singleton.__init__(FormHandler)
     end
 
-    it "allows changing answer on remaining original log" do
-      click_link("Keep this log and delete duplicates", href: "/sales-logs/#{sales_log.id}/delete-duplicates?original_log_id=#{sales_log.id}")
-      click_button "Delete this log"
-      click_link("Change", href: "/sales-logs/#{sales_log.id}/purchaser-code?original_log_id=#{sales_log.id}&referrer=interruption_screen")
-      click_button("Save and continue")
-      expect(page).to have_current_path("/sales-logs/#{sales_log.id}/duplicate-logs?original_log_id=#{sales_log.id}")
-      expect(page).to have_link("Back to Log #{sales_log.id}", href: "/sales-logs/#{sales_log.id}")
-    end
+    context "and updating duplicate log" do
```

Comment:
I think cause the `duplicate_log` gets created before `before` runs. I could have maybe set the date on duplicate log instead, but it would eventually start failing the collection year validation